tools/clang/test/CodeGenDXIL/hlsl/shaders/node/mesh-node-no-invalid-input-record.hlsl

```diff
@@ -0,0 +1,65 @@
+// RUN: %dxc -Tlib_6_9 %s -verify 
+
+// Make sure invalid input records aren't allowed for mesh node shaders
+
+
+struct MY_MATERIAL_RECORD
+{
+    uint textureIndex;
+    float3 normal;
+};
+
+[Shader("node")]
+[OutputTopology("line")]
+// expected-note@+1{{Launch type defined here}}
+[NodeLaunch("mesh")]
+[NumThreads(4,5,6)]
+[NodeDispatchGrid(2,2,2)]
+void myFancyNode1(
+    // expected-error@+1{{'ThreadNodeInputRecord' may not be used with mesh nodes (only DispatchNodeInputRecord)}}
```

Comment:
Seems the diagnostic needs adjustment, since RWDispatchNodeinputRecord is allowed as well, maybe just change it to `[RW]DispatchNodeinputRecord`?

Reply:
And is there a test case that verifies that the RW case is valid?

Reply:
I agree. A test for globallycoherent too since it seems that was allowed as well.